File: llm-platform-starter-sets/typescript-starter/src/core/configLoader.ts
/**
 * configLoader.ts — Reads client and module JSON config files.
 */

import * as fs from "fs";
import * as path from "path";
import { ClientConfig, ModuleConfig } from "./types";

const CONFIG_DIR = path.resolve(__dirname, "../../config");

export function loadClient(clientId: string): ClientConfig {
  const filePath = path.join(CONFIG_DIR, "clients", `${clientId}.json`);
  if (!fs.existsSync(filePath)) {
    throw new Error(`Client config not found: ${filePath}`);
  }
  const raw = fs.readFileSync(filePath, "utf-8");
  return JSON.parse(raw) as ClientConfig;
}

export function loadModuleConfig(moduleName: string): ModuleConfig {
  const filePath = path.join(CONFIG_DIR, "modules", `${moduleName}.json`);
  if (!fs.existsSync(filePath)) {
    throw new Error(`Module config not found: ${filePath}`);
  }
  const raw = fs.readFileSync(filePath, "utf-8");
  return JSON.parse(raw) as ModuleConfig;
}

export function getClientTone(client: ClientConfig): string {
  return client.branding?.tone ?? "clear and professional";
}

export function getClientPermissions(client: ClientConfig, moduleName: string): string[] {
  return client.permissions?.[moduleName] ?? [];
}

File: llm-platform-starter-sets/python-starter/app/core/config_loader.py
"""
config_loader.py — Reads client and module JSON config files.
"""

import json
from pathlib import Path
from typing import Dict, Any
from app.core.types import ClientConfig, ModuleConfig


CONFIG_DIR = Path(__file__).parent.parent.parent / "config"


def load_client(client_id: str) -> ClientConfig:
    """Load a client config by client_id."""
    path = CONFIG_DIR / "clients" / f"{client_id}.json"
    if not path.exists():
        raise FileNotFoundError(f"Client config not found: {path}")
    with open(path) as f:
        data = json.load(f)
    return ClientConfig(data)


def load_module_config(module_name: str) -> ModuleConfig:
    """Load a module config by module name."""
    path = CONFIG_DIR / "modules" / f"{module_name}.json"
    if not path.exists():
        raise FileNotFoundError(f"Module config not found: {path}")
    with open(path) as f:
        data = json.load(f)
    return ModuleConfig(data)
